src/cnc/flavors/aws/shared/provision/partials/collection/database/rds.tf.j2

```diff
@@ -0,0 +1,239 @@
+    )
+
+    auth {
+        auth_scheme = "SECRETS"
+        secret_arn = {% if resource.settings.use_existing %}data.{% endif %}aws_secretsmanager_secret.{{ resource.instance_name }}_db_creds.arn
+        iam_auth = "DISABLED"
+    }
+
+    depends_on = [
+        aws_iam_role.{{ resource.instance_name }}_db_proxy,
+        aws_iam_role_policy.{{ resource.instance_name }}_db_proxy
+    ]
+
+    provisioner "local-exec" {
+        command = "sleep 60"
+    }
+}
+
+          resource "aws_db_proxy_default_target_group" "{{ resource.instance_name }}" {
+            db_proxy_name = aws_db_proxy.{{ resource.instance_name }}.name
+          }
+
+          resource "aws_db_proxy_target" "{{ resource.instance_name }}" {
+            {%- if resource.settings.cluster_mode %}
+              db_cluster_identifier = "{{ resource.settings.identifier }}"
+            {%- else %}
+              db_instance_identifier = "{{ resource.settings.identifier }}"
+            {%- endif %}
+            db_proxy_name     = aws_db_proxy.{{ resource.instance_name }}.name
+            target_group_name = aws_db_proxy_default_target_group.{{ resource.instance_name }}.name
+          }
+
+          data "aws_iam_policy_document" "{{ resource.instance_name }}_db_proxy_service_role" {
+            statement {
+              effect  = "Allow"
+              actions = ["sts:AssumeRole"]
+              principals {
+                type        = "Service"
+                identifiers = ["rds.amazonaws.com"]
+              }
+            }
+          }
+
+          resource "aws_iam_role" "{{ resource.instance_name }}_db_proxy" {
+            name               = "{{ resource.instance_name }}-db-proxy"
+            assume_role_policy = "${data.aws_iam_policy_document.{{ resource.instance_name }}_db_proxy_service_role.json}"
+            provisioner "local-exec" {
+              command = "sleep 30"
+            }
+          }
+
+          data "aws_iam_policy_document" "{{ resource.instance_name }}_db_proxy_service_policy" {
+            statement {
+              effect    = "Allow"
+              resources = ["*"]
+              actions   = ["secretsManager:GetSecretValue"]
+            }
+          }
+
+          resource "aws_iam_role_policy" "{{ resource.instance_name }}_db_proxy" {
+            name   = "{{ resource.instance_name }}-db-proxy"
+            policy = "${data.aws_iam_policy_document.{{ resource.instance_name }}_db_proxy_service_policy.json}"
+            role   = "${aws_iam_role.{{ resource.instance_name }}_db_proxy.id}"
+          }
+        {% endif %}  # if resource.settings.use_db_proxy
+
+{% if not resource.settings.public %}
+
+resource "aws_db_subnet_group" "{{ resource.instance_name }}" {
+    name = "{{ resource.instance_name }}"
+    description = "RDS subnet group"
+    subnet_ids = concat(aws_subnet.{{environment.collection.instance_name}}_private[*].id)
+}
+{% endif %}
+resource "aws_security_group" "{{ resource.instance_name }}_db_access" {
```

Comment:
This is a collection level resource so shouldn't be in this db resources loop. It will also create a conflict if there are both lambda and backend services.